src/test/java/ComplicatedOperationServiceTest.java

```diff
@@ -0,0 +1,392 @@
+        Mockito.when(bOS.subtract(posANegB.getA(), posANegB.getB())).thenReturn(2);
+        Mockito.when(bOS.multiply(posANegB.getA(), posANegB.getB())).thenReturn(-1);
+        Mockito.when(bOS.divide(posANegB.getA(), posANegB.getB())).thenReturn(-1);
+        assertThat(cOS.resultList(posANegB), is(posNeg));
+    }
+
+    @Test
+    public void resultListNegPos() {
+        Mockito.when(bOS.add(negAPosB.getA(), negAPosB.getB())).thenReturn(0);
+        Mockito.when(bOS.subtract(negAPosB.getA(), negAPosB.getB())).thenReturn(-2);
+        Mockito.when(bOS.multiply(negAPosB.getA(), negAPosB.getB())).thenReturn(-1);
+        Mockito.when(bOS.divide(negAPosB.getA(), negAPosB.getB())).thenReturn(-1);
+        assertThat(cOS.resultList(negAPosB), is(negPos));
+    }
+
+    @Test
+    public void resultListNegNeg() {
+        Mockito.when(bOS.add(negANegB.getA(), negANegB.getB())).thenReturn(-2);
+        Mockito.when(bOS.subtract(negANegB.getA(), negANegB.getB())).thenReturn(0);
+        Mockito.when(bOS.multiply(negANegB.getA(), negANegB.getB())).thenReturn(1);
+        Mockito.when(bOS.divide(negANegB.getA(), negANegB.getB())).thenReturn(1);
+        assertThat(cOS.resultList(negANegB), is(negNeg));
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void resultListPosZero() {
+        Mockito.when(bOS.add(posAZeroB.getA(), posAZeroB.getB())).thenReturn(1);
+        Mockito.when(bOS.subtract(posAZeroB.getA(), posAZeroB.getB())).thenReturn(1);
+        Mockito.when(bOS.multiply(posAZeroB.getA(), posAZeroB.getB())).thenReturn(0);
+        Mockito.when(bOS.divide(posAZeroB.getA(), posAZeroB.getB())).thenThrow(new IllegalArgumentException());
+        assertThat(cOS.resultList(posAZeroB), is("Wrong argument - b cannot be 0"));
+    }
+
+    @Test
+    public void resultListZeroPos() {
+        Mockito.when(bOS.add(zeroAPosB.getA(), zeroAPosB.getB())).thenReturn(1);
+        Mockito.when(bOS.subtract(zeroAPosB.getA(), zeroAPosB.getB())).thenReturn(-1);
+        Mockito.when(bOS.multiply(zeroAPosB.getA(), zeroAPosB.getB())).thenReturn(0);
+        Mockito.when(bOS.divide(zeroAPosB.getA(), zeroAPosB.getB())).thenReturn(0);
+        assertThat(cOS.resultList(zeroAPosB), is(zeroPos));
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void resultListZeroZero() {
+        Mockito.when(bOS.add(zeroAZeroB.getA(), zeroAZeroB.getB())).thenReturn(0);
+        Mockito.when(bOS.subtract(zeroAZeroB.getA(), zeroAZeroB.getB())).thenReturn(0);
+        Mockito.when(bOS.multiply(zeroAZeroB.getA(), zeroAZeroB.getB())).thenReturn(0);
+        Mockito.when(bOS.divide(zeroAZeroB.getA(), zeroAZeroB.getB())).thenThrow(new IllegalArgumentException());
+        assertThat(cOS.resultList(zeroAZeroB), is("Wrong argument - b cannot be 0"));
+    }
+
+    //getBasicOperationService Test
+    @Test
+    public void getBasicOperationServiceTest() {
+        assertThat(cOS.getBasicOperationService(), is(bOS));
+    }
```

Comment:
no need to test setters and getters, especially when they're just generated automatically :)